hyperspy/misc/eds/utils.py

```diff
@@ -374,36 +376,41 @@ def quantification_cliff_lorimer(intensities,
     """
     # Value used as an threshold to prevent using zeros as denominator
     min_intensity = 0.1
-    dim = intensities.shape
-    if len(dim) > 1:
-        dim2 = reduce(lambda x, y: x * y, dim[1:])
-        intens = intensities.reshape(dim[0], dim2)
+    original_shape = intensities.shape # (elements, *navigation_shape)
+    if len(original_shape) > 1: # if signal has navigation shape
+        n_elements = intensities.shape[0]
+        n_pixels = np.prod(original_shape[1:])
+        intens = intensities.reshape((n_elements, n_pixels))
         intens = intens.astype('float')
-        for i in range(dim2):
+        for i in range(n_pixels):
             index = np.where(intens[:, i] > min_intensity)[0]
             if len(index) > 1:
+                # Due to k-factors, we only need to compare any two elements
```

Comment:
This wording does not make sense - "k-factors" is not a reason so something cannot be due to it. Please re-word this comment.